lib/Installation/DiskActivation/DiskActivationPage.pm

```diff
@@ -0,0 +1,44 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: The module provides interface to act on the Disk Activation page
+
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package Installation::DiskActivation::DiskActivationPage;
+use parent 'Installation::Navigation::NavigationBase';
+use strict;
+use warnings;
+
+sub init {
+    my ($self, $args) = @_;
+    $self->SUPER::init($args);
+    $self->{btn_conf_dasd} = $self->{app}->button({id => 'dasd'});
+    $self->{btn_conf_zfcp} = $self->{app}->button({id => 'zfcp'});
+    $self->{btn_conf_iscsi} = $self->{app}->button({id => 'iscsi'});
+    return $self;
+}
+
+sub is_shown {
+    my ($self) = @_;
+    return $self->{btn_conf_zfcp}->exist();
+}
```

Comment:
There seems to be a small issue with this `is_shown` because in some cases `btn_conf_zfcp` does not exist, like in [this case](https://openqa.suse.de/tests/7749086#step/configure_dasd/1), which results in test failure.
I think one good way to mitigate this is to remove this `is_shown` function completely, since it is inherited either way from the `NavigationBase` parent, which is checking if the next button exists for it's `is_shown` implementation.
[Here](https://openqa.suse.de/tests/7749121#step/configure_dasd/1) is a small verification run of how removing the `is_shown` function in `DiskActivationPage.pm` seems to work fine.

Reply:
I partly agree with you. Yes, the current problem is that `btn_conf_zfcp` does not exist in your case. Removing the `is_shown()` method in `DiskActivationPage.pm` would then use the `is_shown()` from `NavigationBase.pm`.

What then worries me is that `is_shown()` will be true for **any page** that has a `Next` button. So if e.g. pressing the  `btn_config_zfcp` in my test would not work and I would not end up in the "Configured ZFCP Devices" page I would not notice it if I'm just checking for a `Next` button. On the other hand we don't have a libYUI widget that for "YWizard" that could use the debug_label to determine which page is displayed. 

Of course I have no idea how big the risk is to have problems because of that. So for the moment I'm going to remove the `is_shown()` from ` DiskActivationPage.pm`.

Reply:
You have a valid point here, checking for `Next` button is not a very good check because many pages during installation have a next button so we can not be sure that the page we are interested is indeed shown.

Maybe an implementation that does not include the zfcp button then?